c7n/resources/ecs.py

```diff
@@ -562,6 +584,41 @@ class TaskSubnetFilter(net_filters.SubnetFilter):
     RelatedIdsExpression = "attachments[].details[?name == 'subnetId'].value[]"
 
 
+@Task.filter_registry.register('security-group')
+class TaskSGFilter(net_filters.SecurityGroupFilter):
+
+    RelatedIdsExpression = ""
+    eni_expression = "attachments[].details[?name == 'networkInterfaceId'].value[]"
+    sg_expression = "Groups[].GroupId[]"
+
+    def get_related_ids(self, resources):
+        group_ids = set()
+        eni_ids = set()
+
+        cexp = jmespath_compile(self.eni_expression)
+        for r in resources:
+            ids = cexp.search(r)
+            if ids:
+                eni_ids.update(ids)
+
+        if eni_ids:
+            client = local_session(self.manager.session_factory).client('ec2')
```

Comment:
we might need to cache this, get_related_ids gets called multiple times (first time is with full set of resources afaicr).